source/DataNode.cpp

```diff
@@ -221,6 +221,44 @@ bool DataNode::IsNumber(const string &token)
 
 
 
+// Convert the token at the given index to a boolean. This returns false
+// and prints an error if the index is out of range or the token cannot
+// be interpreted as a number.
+bool DataNode::BoolValue(int index) const
+{
+	// Check for empty strings and out-of-bounds indices.
+	if(static_cast<size_t>(index) >= tokens.size() || tokens[index].empty())
+		PrintTrace("Error: Requested token index (" + to_string(index) + ") is out of bounds:");
+	else if(!IsBool(tokens[index]))
+		PrintTrace("Error: Cannot convert value \"" + tokens[index] + "\" to a boolean:");
+	else
+		return tokens[index] == "true";
+
+	return 0.;
+}
+
+
+
+// Check if the token at the given index is a boolean, i.e. "true" or "false"
+// as a string.
+bool DataNode::IsBool(int index) const
+{
+	// Make sure this token exists and is not empty.
+	if(static_cast<size_t>(index) >= tokens.size() || tokens[index].empty())
+		return false;
+
+	return IsBool(tokens[index]);
+}
+
+
+
+bool DataNode::IsBool(const std::string &token)
+{
+	return token == "true" || token == "false";
```

Comment:
Should we also accept the use of `1` and `0` instead of `true` and `false`?

Reply:
I could see that being done. In the OP, I mentioned this future enhancement:
> Create prefixed conditions that allow for the querying of the states of gamerules. 

Querying a boolean rule would have to return 1 or 0, since conditions are ints. So in that sense, it might make sense to support 1 and 0. But in another sense, I only expect boolean gamerules to be defined with "true" or "false" as their values, not "1" or "0," so is there any reason aside from symmetry with conditions that we'd do support that?

Reply:
I was thinking about changing existing code to use these booleans (for example, preferences could use that instead of 0 and 1 like they do currently for some). If we were to do that we'd of course need to parse 0/1 as well to maintain backwards compatibility.

Reply:
Added 1 and 0 as valid forms of true and false.